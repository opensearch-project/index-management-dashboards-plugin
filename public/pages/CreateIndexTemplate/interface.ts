/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import { RouteComponentProps } from "react-router-dom";
import { FieldInstance } from "../../lib/field";

export interface TemplateDetailProps {
  templateName?: string;
  onCancel?: () => void;
  onSubmitSuccess?: (templateName: string) => void;
  readonly?: boolean;
  history: RouteComponentProps["history"];
}

export interface SubDetailProps extends TemplateDetailProps {
  field: FieldInstance;
  isEdit: boolean;
}
