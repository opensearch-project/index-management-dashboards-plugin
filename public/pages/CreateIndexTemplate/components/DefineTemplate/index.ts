/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import DefineTemplate from "./DefineTemplate";
export { default as OverviewTemplate } from "./OverviewTemplate";

export default DefineTemplate;
