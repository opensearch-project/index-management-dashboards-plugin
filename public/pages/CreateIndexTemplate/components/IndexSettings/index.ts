/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import IndexSettings from "./IndexSettings";

export default IndexSettings;
