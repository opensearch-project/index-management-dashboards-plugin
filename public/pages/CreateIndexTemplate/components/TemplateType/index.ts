/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import TemplateType from "./TemplateType";
export { TemplateConvert } from "./TemplateType";

export default TemplateType;
