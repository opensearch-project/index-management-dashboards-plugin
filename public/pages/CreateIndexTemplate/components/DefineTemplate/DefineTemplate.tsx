/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React from "react";
import {
  EuiCallOut,
  EuiCheckableCard,
  EuiFlexGroup,
  EuiFlexItem,
  EuiPanel,
  EuiSpacer,
  EuiText,
  EuiTextColor,
  EuiHorizontalRule,
} from "@elastic/eui";
import { FLOW_ENUM, SubDetailProps } from "../../interface";
import CustomFormRow from "../../../../components/CustomFormRow";
import { AllBuiltInComponents } from "../../../../components/FormGenerator";
import RemoteSelect from "../../../../components/RemoteSelect";
import { TEMPLATE_NAMING_MESSAGE, TEMPLATE_NAMING_PATTERN } from "../../../../utils/constants";
import TemplateType from "../TemplateType";
import { getCommonFormRowProps } from "../../hooks";
import { filterByMinimatch } from "../../../../../utils/helper";
import { TemplateItem } from "../../../../../models/interfaces";

export default function DefineTemplate(props: SubDetailProps) {
  const { readonly, field, isEdit } = props;
  const values: TemplateItem = field.getValues();
  const Component = isEdit ? AllBuiltInComponents.Text : AllBuiltInComponents.Input;
  const matchSystemIndex = filterByMinimatch(".kibana", values.index_patterns || []);
  const registeredFlowField = field.registerField({
    name: ["_meta", "flow"],
  });
  return readonly ? null : (
    <EuiPanel>
      <EuiText size="s">
        <h2>Template settings</h2>
      </EuiText>
      <EuiHorizontalRule margin="xs" />
      <EuiSpacer size="s" />
      {isEdit ? null : (
        <>
          <CustomFormRow
            {...getCommonFormRowProps("name", field)}
            label={
              <EuiText size="s">
                <h3>Template name</h3>
              </EuiText>
            }
            direction="ver"
            helpText={<div>Template name cannot be changed after the template is created.</div>}
          >
            <Component
              {...field.registerField({
                name: "name",
                rules: [
                  {
                    pattern: TEMPLATE_NAMING_PATTERN,
                    message: "Invalid template name.",
                  },
                ],
              })}
            />
          </CustomFormRow>
          <CustomFormRow helpText={<div>{TEMPLATE_NAMING_MESSAGE}</div>}>
            <></>
          </CustomFormRow>
          <EuiSpacer />
        </>
      )}
      <CustomFormRow
        direction={isEdit ? "hoz" : "ver"}
        {...getCommonFormRowProps("data_stream", field)}
        label={
          <EuiText size="s">
            <h3>Template type</h3>
          </EuiText>
        }
      >
        <TemplateType
          {...field.registerField({
            name: "data_stream",
          })}
        />
      </CustomFormRow>
      <EuiSpacer />
      <CustomFormRow
        {...getCommonFormRowProps("index_patterns", field)}
        label={
          <EuiText size="s">
            <h3>Index patterns</h3>
          </EuiText>
        }
        direction={isEdit ? "hoz" : "ver"}
        helpText="Specify the index patterns or wildcards. Add a comma to separate each value. Settings in this template will be applied to indexes with names matching index patterns or wildcards."
      >
        <RemoteSelect
          {...field.registerField({
            name: "index_patterns",
            rules: [
              {
                validator(rule, value) {
                  if (!value || !value.length) {
                    return Promise.reject("Index patterns must be defined");
                  }

                  return Promise.resolve("");
                },
              },
            ],
          })}
          delimiter=","
          noSuggestions
          async={false}
          refreshOptions={() =>
            Promise.resolve({
              ok: true,
              response: [],
            })
          }
          placeholder="Select index patterns or input wildcards"
        />
      </CustomFormRow>
      <EuiSpacer />
      {matchSystemIndex ? (
        <>
          <CustomFormRow>
            <EuiCallOut color="warning" title="Index patterns may contain system indexes">
              This template may apply to new system indexes and may affect your ability to access OpenSearch. We recommend narrowing your
              index patterns.
            </EuiCallOut>
          </CustomFormRow>
          <EuiSpacer />
        </>
      ) : null}
      <CustomFormRow
        direction={isEdit ? "hoz" : "ver"}
        {...getCommonFormRowProps("priority", field)}
        label={
          <EuiText size="s">
            <h3>Priority</h3>
          </EuiText>
        }
        helpText="Specify the priority of this template. If the index name matches more than one template, the template with the highest priority will be applied to the index."
      >
        <AllBuiltInComponents.Number
          {...field.registerField({
            name: "priority",
            rules: [
              {
                min: 0,
                message: "Priority cannot be smaller than 0.",
              },
              {
                validator(rule, value) {
                  if (Number(value) !== parseInt(value)) {
                    return Promise.reject("Priority must be an integer.");
                  }

                  return Promise.resolve("");
                },
              },
            ],
          })}
        />
      </CustomFormRow>
      <EuiSpacer />
      <CustomFormRow
        {...getCommonFormRowProps(["_meta", "flow"], field)}
        fullWidth
        label={
          <EuiText size="s">
            <h3>Choose a method to define your templates</h3>
          </EuiText>
        }
      >
        <EuiFlexGroup>
          <EuiFlexItem style={{ width: 350 }} grow={false}>
            <EuiCheckableCard
              className="eui-fullHeight"
              label={
                <>
                  <div className="euiCheckableCard__label" style={{ paddingRight: 0 }}>
                    <EuiText size="s">
                      <h4>Simple template</h4>
                    </EuiText>
                  </div>
                  <EuiText size="xs" className="euiCheckableCard__children">
                    <EuiTextColor color="subdued">Define an index template with index aliases, settings, and mappings.</EuiTextColor>
                  </EuiText>
                </>
              }
              id="checkboxForIndexTemplateFlowSimple"
              onChange={() => {
                registeredFlowField.onChange(FLOW_ENUM.SIMPLE);
              }}
              checked={registeredFlowField.value === FLOW_ENUM.SIMPLE}
            />
          </EuiFlexItem>
          <EuiFlexItem style={{ width: 350 }} grow={false}>
            <EuiCheckableCard
              className="eui-fullHeight"
              label={
                <>
                  <div className="euiCheckableCard__label" style={{ paddingRight: 0 }}>
                    <EuiText size="s">
                      <h4>Component template</h4>
                    </EuiText>
                  </div>
                  <EuiText size="xs" className="euiCheckableCard__children">
                    <EuiTextColor color="subdued">
                      Define an index template by associating component templates containing index configurations.
                    </EuiTextColor>
                  </EuiText>
                </>
              }
              id="checkboxForIndexTemplateFlowComponents"
              onChange={() => registeredFlowField.onChange(FLOW_ENUM.COMPONENTS)}
              checked={registeredFlowField.value === FLOW_ENUM.COMPONENTS}
            />
          </EuiFlexItem>
        </EuiFlexGroup>
      </CustomFormRow>
    </EuiPanel>
  );
}
