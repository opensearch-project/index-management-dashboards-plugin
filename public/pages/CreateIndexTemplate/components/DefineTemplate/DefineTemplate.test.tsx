/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { act, render, waitFor } from "@testing-library/react";
import { Route, HashRouter as Router, Switch, Redirect } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import { browserServicesMock, coreServicesMock } from "../../../../../test/mocks";
import DefineTemplate from "./DefineTemplate";
import { ServicesContext } from "../../../../services";
import { CoreServicesContext } from "../../../../components/core_services";
import { ROUTES } from "../../../../utils/constants";
import useField from "../../../../lib/field";
import { FLOW_ENUM, SubDetailProps } from "../../interface";

const WrappedDefineTemplate = (props: Omit<SubDetailProps, "field"> & { onSubmit?: (value: any) => void }) => {
  const field = useField();
  return (
    <>
      <DefineTemplate field={field} {...props} />
      <button
        data-test-subj="submit"
        onClick={async () => {
          const { errors } = await field.validatePromise();
          if (errors) {
            return;
          }
          props.onSubmit?.(field.getValues());
        }}
      >
        submit
      </button>
    </>
  );
};

function renderWithRouter(props: Omit<SubDetailProps, "history" | "field"> & { onSubmit?: (value: any) => void }) {
  return {
    ...render(
      <CoreServicesContext.Provider value={coreServicesMock}>
        <ServicesContext.Provider value={browserServicesMock}>
          <Router>
            <Switch>
              <Route path={ROUTES.COMPOSABLE_TEMPLATES} render={(routeProps) => <WrappedDefineTemplate {...routeProps} {...props} />} />
              <Redirect from="/" to={ROUTES.COMPOSABLE_TEMPLATES} />
            </Switch>
          </Router>
        </ServicesContext.Provider>
      </CoreServicesContext.Provider>
    ),
  };
}

describe("<ComposableTemplatesActions /> spec", () => {
  beforeEach(() => {
    browserServicesMock.commonService.apiCaller = jest.fn(
      async (): Promise<any> => {
        return {
          ok: true,
          response: {
            index_templates: [],
          },
        };
      }
    );
  });
  it("renders the component in non-edit mode", async () => {
    const onChangeMock = jest.fn();
    const { container, getByTestId, findByText } = renderWithRouter({
      isEdit: false,
      onSubmit: onChangeMock,
    });
    await waitFor(() => {
      expect(document.querySelector(".euiComboBox__inputWrap-isLoading")).toBeNull();
    });
    expect(container).toMatchSnapshot();
    await userEvent.click(getByTestId("submit"));
    await findByText("Index patterns must be defined");
    await userEvent.click(container.querySelector("#checkboxForIndexTemplateFlowSimple")?.parentNode as Element);
    await waitFor(() => {
      // eslint-disable-next-line @typescript-eslint/no-unused-expressions
      expect((container.querySelector("#checkboxForIndexTemplateFlowSimple") as HTMLInputElement).checked).toBeTruthy;
      expect(onChangeMock).toBeCalledTimes(0);
    });
    await act(async () => {
      await userEvent.type(getByTestId("form-row-name").querySelector("input") as Element, "1");
      await userEvent.type(getByTestId("form-row-priority").querySelector("input") as Element, "1");
      await userEvent.type(getByTestId("form-row-index_patterns").querySelector("input") as Element, ".kibana*{enter}");
      await userEvent.click(container.querySelector("#checkboxForIndexTemplateFlowComponents") as Element);
      await userEvent.click(getByTestId("submit"));
    });
    await findByText("Index patterns may contain system indexes");
    await waitFor(() => {
      expect(document.querySelector(".euiFormLabel-isInvalid")).not.toBeInTheDocument();
    });
    await waitFor(() => {
      expect(onChangeMock).toBeCalledWith({
        name: "1",
        index_patterns: [".kibana*"],
        priority: "1",
        _meta: {
          flow: FLOW_ENUM.COMPONENTS,
        },
      });
      expect((container.querySelector("#checkboxForIndexTemplateFlowComponents") as HTMLInputElement).checked).toBeTruthy();
    });
  });

  it("renders the component in edit mode", async () => {
    const onChangeMock = jest.fn();
    const { container } = renderWithRouter({
      isEdit: true,
      onSubmit: onChangeMock,
    });
    await waitFor(() => {
      expect(document.querySelector(".euiComboBox__inputWrap-isLoading")).toBeNull();
    });
    expect(container).toMatchSnapshot();
  });

  it("renders the component in readonly mode", async () => {
    const { container } = renderWithRouter({
      isEdit: true,
      readonly: true,
    });
    expect(container).toMatchSnapshot();
  });
});
