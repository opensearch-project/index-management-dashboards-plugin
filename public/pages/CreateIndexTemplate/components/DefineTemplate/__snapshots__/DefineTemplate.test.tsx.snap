// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComposableTemplatesActions /> spec renders the component in edit mode 1`] = `
<div>
  <div
    class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsFlexStart euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h1
          class="euiTitle euiTitle--small"
        >
          Template settings
          <span
            class="panel-header-count"
          >
             
             
          </span>
        </h1>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: 0px 10px;"
    >
      <div
        class="euiSpacer euiSpacer--s"
      />
      <div
        class="euiDescribedFormGroup euiDescribedFormGroup--fullWidth"
        role="group"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
            style="max-width: 700px;"
          >
            <div
              class="euiTitle euiTitle--xsmall euiDescribedFormGroup__title"
            >
              Template type
            </div>
          </div>
          <div
            class="euiFlexItem euiDescribedFormGroup__fields euiDescribedFormGroup__fieldPadding--xsmall"
          >
            <div
              class="euiFormRow"
              data-test-subj="form-row-data_stream"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  class="euiRadio"
                >
                  <input
                    checked=""
                    class="euiRadio__input"
                    id="Indexes"
                    type="radio"
                    value=""
                  />
                  <div
                    class="euiRadio__circle"
                  />
                  <label
                    class="euiRadio__label"
                    for="Indexes"
                  >
                    Indexes
                  </label>
                </div>
                <div
                  class="euiRadio"
                >
                  <input
                    class="euiRadio__input"
                    id="Data streams"
                    type="radio"
                    value=""
                  />
                  <div
                    class="euiRadio__circle"
                  />
                  <label
                    class="euiRadio__label"
                    for="Data streams"
                  >
                    Data streams
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiDescribedFormGroup euiDescribedFormGroup--fullWidth"
        role="group"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
            style="max-width: 700px;"
          >
            <div
              class="euiTitle euiTitle--xsmall euiDescribedFormGroup__title"
            >
              Index patterns
            </div>
            <div
              class="euiText euiText--small euiDescribedFormGroup__description"
            >
              <div
                class="euiTextColor euiTextColor--subdued"
              >
                Specify the index patterns or wildcards. Add a comma to separate each value. Settings in this template will be applied to indexes with names matching index patterns or wildcards.
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiDescribedFormGroup__fields euiDescribedFormGroup__fieldPadding--xsmall"
          >
            <div
              class="euiFormRow"
              data-test-subj="form-row-index_patterns"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="euiComboBox"
                  role="combobox"
                >
                  <div
                    class="euiFormControlLayout"
                  >
                    <div
                      class="euiFormControlLayout__childrenWrapper"
                    >
                      <div
                        class="euiComboBox__inputWrap euiComboBox__inputWrap-isClearable"
                        data-test-subj="comboBoxInput"
                        tabindex="-1"
                      >
                        <p
                          class="euiComboBoxPlaceholder"
                        >
                          Select index patterns or input wildcards
                        </p>
                        <div
                          class="euiComboBox__input"
                          style="font-size: 14px; display: inline-block;"
                        >
                          <input
                            aria-controls=""
                            data-test-subj="comboBoxSearchInput"
                            id="some_html_id"
                            role="textbox"
                            style="box-sizing: content-box; width: 2px;"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiDescribedFormGroup euiDescribedFormGroup--fullWidth"
        role="group"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
            style="max-width: 700px;"
          >
            <div
              class="euiTitle euiTitle--xsmall euiDescribedFormGroup__title"
            >
              Priority
            </div>
            <div
              class="euiText euiText--small euiDescribedFormGroup__description"
            >
              <div
                class="euiTextColor euiTextColor--subdued"
              >
                Specify the priority of this template. If the index name matches more than one template, the template with the highest priority will be applied to the index.
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiDescribedFormGroup__fields euiDescribedFormGroup__fieldPadding--xsmall"
          >
            <div
              class="euiFormRow"
              data-test-subj="form-row-priority"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__fieldWrapper"
              >
                <span
                  class="euiToolTipAnchor euiToolTipAnchor--displayBlock"
                >
                  <div
                    class="euiFormControlLayout"
                  >
                    <div
                      class="euiFormControlLayout__childrenWrapper"
                    >
                      <input
                        class="euiFieldNumber"
                        id="some_html_id"
                        type="number"
                        value=""
                      />
                    </div>
                  </div>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFormRow euiFormRow--fullWidth"
        data-test-subj="form-row-_meta.flow"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            Choose a method to define your template
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="width: 350px;"
            >
              <div
                class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard eui-fullHeight"
              >
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
                >
                  <div
                    class="euiRadio euiRadio--noLabel"
                  >
                    <input
                      class="euiRadio__input"
                      id="checkboxForIndexTemplateFlowSimple"
                      type="radio"
                      value=""
                    />
                    <div
                      class="euiRadio__circle"
                    />
                  </div>
                </div>
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
                >
                  <label
                    class="euiCheckableCard__label"
                    for="checkboxForIndexTemplateFlowSimple"
                  >
                    <div
                      class="euiCheckableCard__label"
                      style="padding-right: 0px;"
                    >
                      Simple template
                    </div>
                    <div
                      class="euiText euiText--extraSmall euiCheckableCard__children"
                    >
                      <span
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Define an index template with index aliases, settings, and mappings.
                      </span>
                    </div>
                  </label>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="width: 350px;"
            >
              <div
                class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard eui-fullHeight"
              >
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
                >
                  <div
                    class="euiRadio euiRadio--noLabel"
                  >
                    <input
                      class="euiRadio__input"
                      id="checkboxForIndexTemplateFlowComponents"
                      type="radio"
                      value=""
                    />
                    <div
                      class="euiRadio__circle"
                    />
                  </div>
                </div>
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
                >
                  <label
                    class="euiCheckableCard__label"
                    for="checkboxForIndexTemplateFlowComponents"
                  >
                    <div
                      class="euiCheckableCard__label"
                      style="padding-right: 0px;"
                    >
                      Component template
                    </div>
                    <div
                      class="euiText euiText--extraSmall euiCheckableCard__children"
                    >
                      <span
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Define an index template by associating component templates containing index configurations.
                      </span>
                    </div>
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    data-test-subj="submit"
  >
    submit
  </button>
</div>
`;

exports[`<ComposableTemplatesActions /> spec renders the component in non-edit mode 1`] = `
<div>
  <div
    class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsFlexStart euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h1
          class="euiTitle euiTitle--small"
        >
          Template settings
          <span
            class="panel-header-count"
          >
             
             
          </span>
        </h1>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: 0px 10px;"
    >
      <div
        class="euiSpacer euiSpacer--s"
      />
      <div
        class="euiFormRow"
        data-test-subj="form-row-name"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            Template name
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFormHelpText euiFormRow__text"
            style="padding-top: 0px; padding-bottom: 4px;"
          >
            <div>
              Template name cannot be changed after the template is created.
            </div>
          </div>
          <span
            class="euiToolTipAnchor euiToolTipAnchor--displayBlock"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  class="euiFieldText"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </span>
        </div>
      </div>
      <div
        class="euiFormRow"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFormHelpText euiFormRow__text"
            style="padding-top: 0px; padding-bottom: 4px;"
          >
            <div>
              Must be in lowercase letters. Cannot begin with underscores. Spaces, commas, and characters *, # are not allowed.
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFormRow"
        data-test-subj="form-row-data_stream"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            Template type
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiRadio"
          >
            <input
              checked=""
              class="euiRadio__input"
              id="Indexes"
              type="radio"
              value=""
            />
            <div
              class="euiRadio__circle"
            />
            <label
              class="euiRadio__label"
              for="Indexes"
            >
              Indexes
            </label>
          </div>
          <div
            class="euiRadio"
          >
            <input
              class="euiRadio__input"
              id="Data streams"
              type="radio"
              value=""
            />
            <div
              class="euiRadio__circle"
            />
            <label
              class="euiRadio__label"
              for="Data streams"
            >
              Data streams
            </label>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFormRow"
        data-test-subj="form-row-index_patterns"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            Index patterns
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFormHelpText euiFormRow__text"
            style="padding-top: 0px; padding-bottom: 4px;"
          >
            Specify the index patterns or wildcards. Add a comma to separate each value. Settings in this template will be applied to indexes with names matching index patterns or wildcards.
          </div>
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            class="euiComboBox"
            role="combobox"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <div
                  class="euiComboBox__inputWrap euiComboBox__inputWrap-isClearable"
                  data-test-subj="comboBoxInput"
                  tabindex="-1"
                >
                  <p
                    class="euiComboBoxPlaceholder"
                  >
                    Select index patterns or input wildcards
                  </p>
                  <div
                    class="euiComboBox__input"
                    style="font-size: 14px; display: inline-block;"
                  >
                    <input
                      aria-controls=""
                      data-test-subj="comboBoxSearchInput"
                      role="textbox"
                      style="box-sizing: content-box; width: 2px;"
                      value=""
                    />
                    <div
                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFormRow"
        data-test-subj="form-row-priority"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            Priority
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFormHelpText euiFormRow__text"
            style="padding-top: 0px; padding-bottom: 4px;"
          >
            Specify the priority of this template. If the index name matches more than one template, the template with the highest priority will be applied to the index.
          </div>
          <span
            class="euiToolTipAnchor euiToolTipAnchor--displayBlock"
          >
            <div
              class="euiFormControlLayout"
            >
              <div
                class="euiFormControlLayout__childrenWrapper"
              >
                <input
                  class="euiFieldNumber"
                  type="number"
                  value=""
                />
              </div>
            </div>
          </span>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--l"
      />
      <div
        class="euiFormRow euiFormRow--fullWidth"
        data-test-subj="form-row-_meta.flow"
        id="some_html_id-row"
      >
        <div
          class="euiFormRow__labelWrapper"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel euiFormRow__label"
            for="some_html_id"
          >
            Choose a method to define your template
          </label>
        </div>
        <div
          class="euiFormRow__fieldWrapper"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="width: 350px;"
            >
              <div
                class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard eui-fullHeight"
              >
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
                >
                  <div
                    class="euiRadio euiRadio--noLabel"
                  >
                    <input
                      class="euiRadio__input"
                      id="checkboxForIndexTemplateFlowSimple"
                      type="radio"
                      value=""
                    />
                    <div
                      class="euiRadio__circle"
                    />
                  </div>
                </div>
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
                >
                  <label
                    class="euiCheckableCard__label"
                    for="checkboxForIndexTemplateFlowSimple"
                  >
                    <div
                      class="euiCheckableCard__label"
                      style="padding-right: 0px;"
                    >
                      Simple template
                    </div>
                    <div
                      class="euiText euiText--extraSmall euiCheckableCard__children"
                    >
                      <span
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Define an index template with index aliases, settings, and mappings.
                      </span>
                    </div>
                  </label>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="width: 350px;"
            >
              <div
                class="euiPanel euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow euiPanel--hasBorder euiPanel--flexGrowZero euiSplitPanel euiSplitPanel--row euiCheckableCard eui-fullHeight"
              >
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--subdued euiPanel--noShadow euiPanel--noBorder euiPanel--flexGrowZero euiPanel--isClickable euiSplitPanel__inner"
                >
                  <div
                    class="euiRadio euiRadio--noLabel"
                  >
                    <input
                      class="euiRadio__input"
                      id="checkboxForIndexTemplateFlowComponents"
                      type="radio"
                      value=""
                    />
                    <div
                      class="euiRadio__circle"
                    />
                  </div>
                </div>
                <div
                  class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusNone euiPanel--transparent euiPanel--noShadow euiPanel--noBorder euiSplitPanel__inner"
                >
                  <label
                    class="euiCheckableCard__label"
                    for="checkboxForIndexTemplateFlowComponents"
                  >
                    <div
                      class="euiCheckableCard__label"
                      style="padding-right: 0px;"
                    >
                      Component template
                    </div>
                    <div
                      class="euiText euiText--extraSmall euiCheckableCard__children"
                    >
                      <span
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Define an index template by associating component templates containing index configurations.
                      </span>
                    </div>
                  </label>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    data-test-subj="submit"
  >
    submit
  </button>
</div>
`;

exports[`<ComposableTemplatesActions /> spec renders the component in readonly mode 1`] = `
<div>
  <button
    data-test-subj="submit"
  >
    submit
  </button>
</div>
`;
